irohad/main/port_check.hpp

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_PORT_CHECK_HPP
+#define IROHA_PORT_CHECK_HPP
+
+#include <netinet/in.h>
+#include <arpa/inet.h>
+
+bool checkTcpPortInUse(std::string address, size_t port) {
+    struct sockaddr_in client;
+    int sock;
+
+    client.sin_family = AF_INET;
+    client.sin_port = htons(port);
+    client.sin_addr.s_addr = inet_addr(address.c_str());
+
+    sock = (int) socket(AF_INET, SOCK_STREAM, 0);
+
+    int result = connect(sock, (struct sockaddr *) &client,sizeof(client));
```

Comment:
Not really the best way to check the port. Port might be allocated, but unused, so connect will be failed. Imo it's better check it with `bind` + SO_REUSEADDR